src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -151,10 +151,30 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
         };
     }
 
-    protected ackDecorator(e: any, messageId: number): AckFunc {
+    private addBreadcrumb(message: any,  messageId: number, action: string, name: string): void {
+        if (message && message.breadcrumbs && message.breadcrumbs.length) {
+            message.breadcrumbs.push({
+                action,
+                messageId,
+                name,
+                time: Date.now()
+            });
+        }
+    }
+
+    protected ackDecorator(e: any, messageId: number, originalPayload: any): AckFunc {
         const ackObj = new AckMessage();
         ackObj.correlationId = messageId;
 
+        // It is inferred that breadcrumbs have been enabled for a window
+        // when the breadcrumb array is present in the original payload of the API request.
+        const usingBreadcrumbs = originalPayload.breadcrumbs && originalPayload.breadcrumbs.length;
```

Comment:
Why do we need to check the length? Shouldn't it be enough that breadcrumbs exists?

Reply:
That's to perform a sanity check against a breadcrumb enabled message originating from an adapter VS a breadcrumb formed envelope as generated in the core